Frontend/src/Components/Graph/CustomGraph.js

```diff
@@ -1,427 +1,496 @@
+        borderCapStyle: "round",
+        borderJoinStyle: "round",
+      },
+    },
+    interaction: {
+      mode: "nearest",
+      axis: "x",
+      intersect: false,
+    },
+    datasets: {
+      line: {
+        pointBackgroundColor: "transparent",
+        pointBorderColor: "transparent",
+        pointRadius: 10,
+      },
+    },
+  };
+}
+
+/**
+ * Creates a customizable graph
+ *
+ * @param {any} props the props to pass to this graph; any {@link StackProps} will be passed to the overarching container
+ * @param {(name: string, isNew: boolean, datasets: string[]) => void} props.onSave the callback function for when the user attempts to save this graph
+ * @param {string} props.title the title that this graph has
+ * @param {{category: string, values: Dataset[]}[]} props.categories the color and other associated data for each dataset, organized in an object
+ * @param {Dataset[]} props.allDatasets a list of all datasets, complete with ID and color information
+ * @param {string[]} props.initialDatasets a list of the IDs of the initial datasets
+ * @param {any} props.packedData the queue of data, packed in Chart.js format, coming from the solar car
+ * @param {number} props.secondsRetained the number of seconds to retain a point on this grpah
+ * @returns
+ */
+export default function CustomGraph(props) {
+  // destructure props
+  const {
+    onSave,
+    title,
+    categories,
+    packedData,
+    initialDatasets,
+    allDatasets,
+    secondsRetained,
+    latestTime,
+    ...stackProps
+  } = props;
+
+  // useEffect(() => {
+  //   console.log(
+  //     packedData
+  //       .find((data) => data.key === "speed")
+  //       .data.map((point) => point.x)
+  //       .join(", ")
+  //   );
+  // }, [packedData]);
+
+  // disclosure for the data selection modal
+  const {
+    isOpen: isDataSelectOpen,
+    onOpen: onDataSelectOpen,
+    onClose: onDataSelectClose,
+  } = useDisclosure();
+  // disclosure for the name modal
+  const {
+    isOpen: isNameModalOpen,
+    onOpen: onNameModalOpen,
+    onClose: onNameModalClose,
+  } = useDisclosure();
+
+  // an object that contains every selected dataset and a boolean to store whether it is enabled
+  const [datasets, setDatasets] = useState(() => {
+    const output = {};
+    for (const dataset of initialDatasets) {
+      output[dataset] = true;
+    }
+
+    // console.log("init datasets:", output, "from", initialDatasets);
+
+    return output;
+  });
+  // useEffect(() => {
+  //   console.log(`datasets for "${title}" updated:`, datasets);
+  // }, [datasets, title]);
+
+  // the data to pass to the graph
+  const [formattedDatasets, setFormattedDatasets] = useState([]);
+  // the keys of the dataset array
+  const [datasetKeys, setDatasetKeys] = useState(Object.keys(datasets));
+  // called when the name is to be saved, memoized for performance
+  const onNameSave = useCallback(
+    (name, isNew) => onSave(name, isNew, datasetKeys),
+    [onSave, datasetKeys]
+  );
```

Comment:
Why pass isNew and not just imply true on line 211 since it's always true in the name modal?